src/Fantomas.Core/Context.fs

```diff
@@ -752,14 +752,31 @@ let sepSpaceOrDoubleIndentAndNlnIfExpressionExceedsPageWidth expr (ctx: Context)
         expr
         ctx
 
-let isStroustrupApplicable isStroustrupContext (node: Node) (ctx: Context) =
-    ctx.Config.ExperimentalStroustrupStyle
-    && isStroustrupContext
-    && Seq.isEmpty node.ContentBefore
-
-let isNamedComputationAndPreferSameLine expr (ctx: Context) =
-    match expr with
-    | Expr.NamedComputation _ when ctx.Config.PreferComputationExpressionNameOnSameLine -> true
+let isStroustrupExpr (config: FormatConfig) (e: Expr) =
+    let isStroustrupEnabled = config.MultilineBracketStyle = ExperimentalStroustrup
+
+    match e with
+    | Expr.Record node when isStroustrupEnabled ->
+        match node.Extra with
+        | RecordNodeExtra.Inherit _
+        | RecordNodeExtra.With _ -> false
+        | RecordNodeExtra.None -> true
+    | Expr.AnonRecord node when isStroustrupEnabled ->
+        match node.CopyInfo with
+        | Some _ -> false
+        | None -> true
+    | Expr.ArrayOrList _ when isStroustrupEnabled -> true
+    | Expr.NamedComputation _ when config.PreferComputationExpressionNameOnSameLine -> true
```

Comment:
I didn't think about it earlier but we can probably have:
`| Expr.NamedComputation _ -> not config.NewlineBeforeMultilineComputationExpression` here instead.